test/unittest/datasets/unittest_random_data_producers.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file data_producer_common_tests.cpp
+ * @date 12 July 2021
+ * @brief Common test for nntrainer dataset producers (Param Tests)
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <gtest/gtest.h>
+
+#include <algorithm>
+
+#include <data_producer_common_tests.h>
+#include <random_data_producers.h>
+
+DataProducerValidatorType random_onehot_validator(float min, float max) {
+  /// input validator: every value is in range of min, max
+  auto input_valid = [min, max](const nntrainer::Tensor &t) {
+    auto data = t.getData();
+    for (unsigned int i = 0; i < t.length(); ++i) {
+      if (*data < min || max < *data) {
+        return false;
+      }
+      data++;
+    }
+    return true;
+  };
+
+  /// label validator: sum of all is equal to batch
+  auto label_valid = [](const nntrainer::Tensor &t) {
+    return fabs(t.batch() - t.sum({0, 1, 2, 3}).getValue(0, 0, 0, 0) < 1e-7);
```

Comment:
This would turn out to be valid if one of the batch values adds to 2 and some other value adds to 0.
Instead, please update the check for each batch, the sum to be 1.

Reply:
Let me change this within the element-wise generator change.

Reply:
added todo instead

Reply:
Please update this before this code/this type of check is used at other places :).

Reply:
this type check is only used inside random producers, other producer will have it's own validator.

Reply:
cool.